src/Components/getList.js

```diff
@@ -27,16 +27,67 @@ function EmptyList() {
   );
 }
 
-function FullList(props) {
+
+const FullList = (props)=> {
   const [isChecked, setCheck] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleChange = e => {
+    setSearchTerm(e.target.value);
+  };
+
+  const clearSearch = () => {
+    setSearchTerm("");
+  };
+
+  const searchList = props.data.filter(item => {
+    return item.itemName.toLowerCase().includes(searchTerm.toLowerCase());
+  });
+
 
 
   return (
     <div className="grocery-list">
       <h1>Groceries</h1>
-      <ul className="tl f2">
-        {props.data.map(item => (
+      <ul className="tl f2 check-list">
+      <li className="bg-white pa1 shadow">
+      <div className="filter-input">
+      <input 
+         type="text"
+         placeholder="Filter List"
+         value={searchTerm}
+         onChange={handleChange}
+       />
+       <button className="filter-button" onClick={clearSearch}> X </button>
+       </div>
+       </li>
+
+    
+        {searchTerm
+          ? searchList.map(item => (
+            <li className="bg-white pa1 shadow" key={item.id}>
+               <div className="checkbox-input">
+            <label>
+                    <input 
+                      type="checkbox"
+                        value= {isChecked}
```

Comment:
I noticed this state is stored for the entire list, while each individual item will need to have its own checked state. I wouldn't fix it here, but we should talk with the team to figure out how to resolve this bug to make sure it doesn't get in our way for future stories!